### PR TITLE
Add Next Meeting

### PR DESCRIPTION
**Pull request by benedictchuajj**
### Description

Add NextMeeting class which encapsulates:
* date
* startTime
* endTime
* location
of a meeting. The prefix of a NextMeeting is `m/`.

NextMeeting information is shown in PersonViewPanel as well as PersonCard.

Format to submit a NextMeeting as a string is:
`m/dd-mm-yyyy (hh:mm~hh:mm), {non-empty location string)`.

For example, a valid NextMeeting is:
`m/25-12-2021 (00:00~23:59), Santa's Workshop`

#### Breaking change (FIXED with 565c2e5):

PersonCard width behaves very differently when NextMeeting string exceeds PersonCard length or is within PersonCard length.

Screenshot 2021-10-18 at 10 41 41 PM

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update
- [ ] Code quality improvements

### How to test

From the sample address book, you can run this command to check out the behaviour of NextMeeting.

Command | Behaviour
-- | --
add n/Carl e/carl@gmail.com | add a Client named Carl with 'No meeting planned'
add n/Nick e/nick@gmail.com m/20-12-2021 (12:00~12:15), Zoom | add a Client named Nick with NextMeeting set up
edit 6 m/18-10-2021 (18:00~19:00), NUS | update Carl's NextMeeting

### Checklist

- [X] I have tested this code
- [x] I have updated the documentation

### CODE REVIEW

**Comment by CrownKira** [comment]
Is it possible to adjust the UI such that the NextMeeting panes all occupy the same width? 🤔

File: src/main/java/seedu/address/model/person/NextMeeting.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    public final String location;
+
+    /**
+     * Constructs a {@code NextMeeting}.
+     *
+     * @param date date agent next meets a client
+     */
+    public NextMeeting(String date, String startTime, String endTime, String location) {
+        if (!IS_NULL_VALUE_ALLOWED) {
+            requireNonNull(date);
```
Comment (CrownKira) [inline src/main/java/seedu/address/model/person/NextMeeting.java:44]: I believe there is a `requireAllNonNull() `

**Review by CrownKira — APPROVED** [review]
LGTM :)

File: src/main/java/seedu/address/model/person/NextMeeting.java (hunk 1 above)
Comment (benedictchuajj) [reply src/main/java/seedu/address/model/person/NextMeeting.java]: Got it! Have changed to this instead of multiple requireNonNull 😅

**Comment by benedictchuajj** [comment]
> Is it possible to adjust the UI such that the NextMeeting panes all occupy the same width? 🤔

I've fixed the NextMeeting panes to use the same width for now! If we need to expand the NextMeeting panes, we can revert this change next time.

File: src/main/java/seedu/address/model/person/NextMeeting.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        + "where Day, month and year should be numerical values.";
+    public static final String TIME_MESSAGE_CONSTRAINTS = "Next meeting time should be in the 24-hour format, "
+        + "where Hour and Minutes should be numerical values.";
+    public static final String MESSAGE_INVALID_MEETING_STRING = "String representation of Next Meeting is not correct";
+    public static final String NO_NEXT_MEETING = "No meeting planned.";
+    public static final NextMeeting NULL_MEETING = new NextMeeting(null, null, null,
+        null);
+
+    public static final String VALID_MEETING_STRING =
+        "([0-9]{2})-([0-9]{2})-([0-9]{4}) \\(([0-9]{2}):([0-9]{2})~([0-9]{2}):([0-9]{2})\\),(.|\\s)*\\S(.|\\s)*";
```
Comment (wyrchris) [inline src/main/java/seedu/address/model/person/NextMeeting.java:26]: The regex here with respect to date and time portion seems to be less strict or different than that of Date/Time regex in StringUtil. Is this perhaps intentional?

**Pull request merged**